### ISSUE TITLE
microsoftobjectclass.schema incomplete

### ISSUE THREAD

**Issue by AdrianVollmer** [dkoudela/active-directory-to-openldap#10]
I was wondering what the source of `microsoftobjectclass.schema` is.

I admit my LDAP skills are quite limited, but it appears that it is incomplete as it does not contain all attributes. For instance, `gPLink` is defined here https://github.com/dkoudela/active-directory-to-openldap/blob/4aa4a74ea8ec05e62eb70065ffb22bb48ace1116/schema/microsoftattributetype.schema#L4920 but missing here https://github.com/dkoudela/active-directory-to-openldap/blob/4aa4a74ea8ec05e62eb70065ffb22bb48ace1116/schema/microsoftobjectclass.schema#L45

Am I misunderstanding something or should the second location not contain a lot more attributes? If Microsoft decides to add more attributes, where could I get a definitive list?

**Comment by stevenlafl** [comment]
Have you tried adding `$ gPLink` to the `MAY` clause in that location under `microsoftobjectclass.schema` ?

**Comment by AdrianVollmer** [comment]
Yes, that works. I have to add dozens of attributes though. I just need to know if that has been an oversight or if I'm doing something wrong. And if there is a reliable source I'd prefer to use that instead of fixing it manually, but if not, I'll do just that.

**Comment by stevenlafl** [comment]
Anything you find there is likely missing for others, or would be in setups
like yours. If you are modifying those files, can you contribute those
types of changes back into a PR? Since they're just alterations I think
referencing this issue would help future folks. Unfortunately the
repository owner appears to be.. absent.

**Comment by AdrianVollmer** [comment]
Sure, if I end up with something useful, I'll create a PR. My use case might be a bit special though, since I'm only interested in read-only access, so I don't care if the schema is 100% correct as long as it is "good enough". Also, I don't work with LDAP schemas a lot, so I'm not entirely sure  that I know what I'm doing. But I'm happy to publish a fork either way.

And if I could get the schema somewhere from Microsoft I think I could save a lot of time and work while getting a better result.
